test/jdk/java/awt/FileDialog/FileNameOverrideTest.java

```diff
@@ -0,0 +1,112 @@
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.FileDialog;
+import java.awt.Toolkit;
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import javax.swing.JFrame;
+import javax.swing.JTextField;
+
+public class FileNameOverrideTest {
+    private final static String fileName = "input";
+    private final static String clickDirName = "Directory for double click";
+    private final static String dirPath = System.getProperty("user.dir");;
+    private static Button showBtn;
+    private static FileDialog fd;
+
+    private static JFrame initialize() {
+        JFrame frame = new JFrame("File Name Override Test Frame");
+        frame.setLayout(new BorderLayout());
+        JTextField textOutput = new JTextField(30);
+        frame.add(textOutput, BorderLayout.NORTH);
+
+        fd = new FileDialog(frame, "Open");
+
+        showBtn = new Button("Show File Dialog");
+        showBtn.addActionListener(w -> {
+            fd.setFile(fileName);
+            fd.setDirectory(dirPath);
+            fd.setVisible(true);
+            String output = fd.getFile();
+            if (fileName.equals(output)) {
+                textOutput.setText("TEST PASSED");
+            } else {
+                textOutput.setText("TEST FAILED (output file - " + output + ")");
+            }
+        });
+        frame.add(showBtn, BorderLayout.SOUTH);
+
+        try {
+            File tmpFileUp = new File(dirPath + File.separator + fileName);
+            File tmpDir = new File(dirPath + File.separator + clickDirName);
+            File tmpFileIn = new File(tmpDir.getAbsolutePath() + File.separator + fileName);
+            tmpDir.mkdir();
+            tmpFileUp.createNewFile();
+            tmpFileIn.createNewFile();
+        } catch (IOException ex) {
+            throw new RuntimeException("Cannot create test folder", ex);
+        }
+
+        frame.pack();
+        return frame;
+    }
+
+    public static void main(String[] args) throws InterruptedException,
+            InvocationTargetException {
+        String instructions = """
+                1) Click on 'Show File Dialog' button. A file dialog will come up.
+                2) Double-click on '$clickDirName' and click OK.
+                3) The text in the text field will indicate if test is passed or failed.
+                """;
+
+        String toolkit = Toolkit.getDefaultToolkit().getClass().getName();
+        if (!toolkit.equals("sun.awt.X11.XToolkit")) {
+            instructions = """
+                    The test is not applicable for $toolkit. Press Pass.
+                    """. replace("$toolkit", toolkit);
+        }
+
+        PassFailJFrame.builder()
+                .title("File Dialog Return Test Instructions")
+                .instructions(instructions)
+                .rows((int) instructions.lines().count() + 1)
+                .columns(50)
+                .testUI(FileNameOverrideTest::initialize)
```

Comment:
It is possible to get rid of the extra frame, e.g. make the `initialize` return the "Show File Dialog" `JButton`

```suggestion
                .splitUIBottom(FileNameOverrideTest::initialize)
```